com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -442,14 +443,15 @@ define([
                 $scope.errors.newStartDate = [];
                 $scope.errors.newEndDate = [];
 
-                $scope.validateDates($scope.edit_data.new_role_id.newStartDate, $scope.edit_data.new_role_id.newEndDate, function (error, field) {
-                    errors++;
-                    if (field.indexOf('start_date') > -1) {
-                        $scope.errors.newStartDate.push(error);
-                    }
-                    if (field.indexOf('end_date') > -1) {
-                        $scope.errors.newEndDate.push(error);
-                    }
+                var contract = me.contractsData[$scope.edit_data.new_role_id.job_contract_id];
+
+                errors = validateDates({
```

Comment:
Also, does `validateDates` need to return the errors count at all? As far as I can see the `errors` value is used only to check if its `> 0` so we might as well have the function return a boolean instead
```js
if (validateDates(/* ... */)) {

} else {

}
```